rfcs/0000-single-cluster-dnspolicy.md

```diff
@@ -0,0 +1,286 @@
+    kind: Gateway
+  strategy: simple
+```
+
+```yaml
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: Gateway
+metadata:
+  name: prod-web
+  namespace: my-gateways
+spec:
+  gatewayClassName: istio
+  listeners:
+    - allowedRoutes:
+        namespaces:
+          from: All
+      name: api
+      hostname: "myapp.mn.hcpapps.net"
+      port: 80
+      protocol: HTTP
+status:
+  addresses:
+    - type: IPAddress
+      value: 172.31.200.0
+```
+
+In the example the `api` listener has a hostname `myapp.mn.hcpapps.net` that matches a hosted zone being managed by the provider referenced `my-route53-credentials` in the DNSPolicy. 
+As the `simple` strategy is set in the DNSPolicy a DNSRecord resource with the following contents will be created:
+
+```yaml
+apiVersion: kuadrant.io/v1alpha1
+kind: DNSRecord
+metadata:
+  name: prod-web-api
+  namespace: my-gateways
+spec:
+  providerRef:
+    name: my-route53-credentials
+    namespace: my-gateways
+  endpoints:
+    - dnsName: myapp.mn.hcpapps.net
+      recordTTL: 60
+      recordType: A
+      targets:
+        - 172.31.200.0
+```
+
+The `providerRef` is included in the DNSRecord to allow the MGC dns record controller to load the appropriate provider during reconciliation and create the DNS records in the dns provider service e.g. route 53.
+
+**Example 2.** DNSPolicy using `simple` strategy with external dns provider
+
+```yaml
+apiVersion: kuadrant.io/v1alpha1
+kind: DNSPolicy
+metadata:
+  name: prod-web
+  namespace: my-gateways
+spec:
+  providerRef:
+    name: external-dns
+    kind: ExternalDNS
+  targetRef:
+    name: prod-web
+    group: gateway.networking.k8s.io
+    kind: Gateway
+  strategy: simple
+```
+
+In ths example if the DNSPolicy was attached to the same gateway described in example 1, the same DNSRecord would also be created but MGC would not reconcile it. In this scenario it is expected that an external controller is being used to manage the reconciliation of the DNSRecord resources such as [external-dns](https://github.com/kubernetes-sigs/external-dns).
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+## API Updates
+
+DNSPolicy:
+
+* - new providerRef field `spec.providerRef` 
+* - new strategy field `spec.strategy` 
+
+DNSRecord:
+
+- `spec.managedZone` replaced with `spec.providerRef`
+
+### DNSPolicy.spec.providerRef
+
+The `providerRef` field is mandatory and contains a reference to a resource that shows how DNSRecords will be reconciled.
+    * - `spec.providerRef.name` - name of the provider resource
+    * - `spec.providerRef.namespace` - namespace of provider resource
+    * - `spec.providerRef.kind` - kind of resource, can be anything but only `Secret` will be reconciled by MGC.
+
+A kind of type `Secret` will be managed by MGC, and a secret in the given namespace with the given name must exist. The expected contents of the secrets data is comparable to the `dnsProviderSecretRef` used by ManageZones. 
```

Comment:
should we align these? Prob move managedzone to use ```providerRef``` also? its  not part of this obv, could just be reflected in an issue

Reply:
Yep, this will be renamed, but in this doc i guess we are talking about the current version of the code which has it as `dnsProviderSecretRef`.